File: rea2/app/src/main/java/com/jang/assistales/model/UniversBean.java
package com.jang.assistales.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.io.Serializable;

@Entity
public class UniversBean implements Serializable {
    @PrimaryKey(autoGenerate = true)
    private long id_univers;
    private String name;

    @Override
    public String toString() {
        return name;
    }

    //--------------------------Constructor---------------------------//

    public UniversBean(String name) {
        this.name = name;
    }


    //--------------------------Getter/Setter-------------------------//


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: rea5/assistales/app/src/main/java/com/jang/assistales/model/UniversBean.kt
package com.jang.assistales.model

import androidx.room.*

@Entity(tableName = "univers")
class UniversBean(
    val name: String,
    @PrimaryKey(autoGenerate = true)
    var id_univers: Long = 0
) {
    override fun toString(): String {
        return name
    }


    @Dao
    interface UniversDao {
        @Query("SELECT * FROM univers ORDER BY name ASC")
        fun getAll(): List<UniversBean>

        @Insert
        fun insert(univers: UniversBean): Long

        @Update
        fun update(univers: UniversBean)

        @Delete
        fun delete(univers: UniversBean)

        @Query("SELECT name FROM univers ORDER BY name ASC")
        fun getName(): List<String>
    }

}